### PR TITLE
Merge S6658 into S935

### PR DESCRIPTION
**Pull request by anton-haubner-sonarsource**
This PR merges S6658 into S935, see also this discussion on slack: https://sonarsource.slack.com/archives/CFUS31LRE/p1690534750237959

This is the previous PR for merging S6658 which this PR is based on: https://github.com/SonarSource/rspec/pull/2450

## Review

A dedicated reviewer checked the rule description successfully for:

- [ ] logical errors and incorrect information
- [ ] information gaps and missing content
- [ ] text style and tone
- [ ] PR summary and labels follow [the guidelines](https://github.com/SonarSource/rspec/#to-modify-an-existing-rule)

### CODE REVIEW

File: rules/S935/python/rule.adoc [hunk 1]
```diff
@@ -48,24 +63,21 @@ ifdef::env-github,rspecator-view[]
...
 === Highlighting
 
-* if the returned type is wrong or nothing should be returned.
-** Primary: the ``++return++`` statement.
-** Secondary: The function/method name.
-* if no value is returned and one is expected.
-** Primary: the function/method name
+==== if there is return statement
+    Primary: return statement
+    Secondary: method name
```
Comment (anton-haubner-sonarsource) [inline rules/S935/python/rule.adoc:73]: The implementation does not mark the method name at all, if there is a return statement.
Should we adapt the implementation to mark a secondary location?

File: rules/S935/python/rule.adoc [hunk 2]
```diff
@@ -48,24 +63,21 @@ ifdef::env-github,rspecator-view[]
...
+
+==== if there is no return statement at all
+    Primary: method name
```
Comment (anton-haubner-sonarsource) [inline rules/S935/python/rule.adoc:76]: The implementation marks the function definition starting from the `def` keyword up until the `:` instead.
Is this an issue?

File: rules/S935/python/rule.adoc [hunk 3]
```diff
@@ -48,24 +63,21 @@ ifdef::env-github,rspecator-view[]
 
 === Message
 
-* Return a value of type "XXX" instead of "YYY", as should any "ZZZ" method.
-* Return "None" instead of a value of type "XXX", as should any "ZZZ" method.
-* Return a value of type "XXX"; "ZZZ" may reach its end and return None.
+* Return a value of type XXX here.
```
[the remark this answers is not in the document]
Comment (andreaguarino) [reply rules/S935/python/rule.adoc:66]: Yes we can update RSPEC, can you please take care of it?

File: rules/S935/python/rule.adoc [hunk 4]
```diff
@@ -48,24 +63,21 @@ ifdef::env-github,rspecator-view[]
...
 
 === Highlighting
 
-* if the returned type is wrong or nothing should be returned.
-** Primary: the ``++return++`` statement.
-** Secondary: The function/method name.
-* if no value is returned and one is expected.
-** Primary: the function/method name
+==== if there is return statement
+    Primary: return statement
```
[the remark this answers is not in the document]
Comment (andreaguarino) [reply rules/S935/python/rule.adoc:72]: In my opinion, it's fine in this case to keep it as is

File: rules/S935/python/rule.adoc (hunk 1 above)
Comment (andreaguarino) [reply rules/S935/python/rule.adoc:73]: Let's create a ticket to add the secondary location and we can handle it in another sprint

File: rules/S935/python/rule.adoc (hunk 2 above)
Comment (andreaguarino) [reply rules/S935/python/rule.adoc:76]: In my opinion, it's fine in this case to keep it as is

**Review by andreaguarino — APPROVED** [review]
LGTM! I let you update the message section before merging

**Pull request merged**